test/compile_errors.zig

```diff
@@ -183,4 +183,16 @@ pub fn addCases(ctx: *Cases) !void {
             ":1:1: note: invalid byte: '\\xff'",
         });
     }
+
+    {
+        const case = ctx.addTest("duplicate test name", .{});
+
+        case.addError(
```

Comment:
This test should be a file in the `test/cases/compile_errors` directory; see `access_invalid_typeInfo_decl.zig` for an example.

The standalone test could also be turned into behavior tests but that's more subjective.